### PR TITLE
First implementation

### PR DESCRIPTION
**Pull request by Katarzyna-Kadziolka**
Complete client

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -686,4 +686,5 @@ fabric.properties
 
 
 # ===== custom =====
-.idea/
\ No newline at end of file
+.idea/
+.runsettings
```
Comment (RinoReyns) [inline .gitignore]: endline is missing :P

File: StabilityClient.Net/StabilityClient.Net.Tests.Unit/ClientTests.cs [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    
+    [Test]
+    public void Constructor_ShouldInitializedProject() {
+        // Arrange
+        // Act
+        StabilityClient stabilityClient = new StabilityClient();
+        // Assert
+        stabilityClient.Project.Should().NotBeNull();
+    }
+}
```
Comment (RinoReyns) [inline StabilityClient.Net/StabilityClient.Net.Tests.Unit/ClientTests.cs]: endline is missing

File: StabilityClient.Net/StabilityClient.Net.Tests.Unit/GlobalUsings.cs [hunk 3]
```diff
@@ -0,0 +1,4 @@
+// Global using directives
+
+global using NUnit.Framework;
+global using FluentAssertions;
```
Comment (RinoReyns) [inline StabilityClient.Net/StabilityClient.Net.Tests.Unit/GlobalUsings.cs]: endline is missing

File: StabilityClient.Net/StabilityClient.Net.sln [hunk 4]
```diff
@@ -12,5 +14,9 @@ Global
 		{4F5ABEB5-02DC-436F-A736-0BC23AEA2F33}.Debug|Any CPU.Build.0 = Debug|Any CPU
 		{4F5ABEB5-02DC-436F-A736-0BC23AEA2F33}.Release|Any CPU.ActiveCfg = Release|Any CPU
 		{4F5ABEB5-02DC-436F-A736-0BC23AEA2F33}.Release|Any CPU.Build.0 = Release|Any CPU
+		{23A67BFD-D323-430D-8FE1-14CC58B4D2EA}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
+		{23A67BFD-D323-430D-8FE1-14CC58B4D2EA}.Debug|Any CPU.Build.0 = Debug|Any CPU
+		{23A67BFD-D323-430D-8FE1-14CC58B4D2EA}.Release|Any CPU.ActiveCfg = Release|Any CPU
+		{23A67BFD-D323-430D-8FE1-14CC58B4D2EA}.Release|Any CPU.Build.0 = Release|Any CPU
 	EndGlobalSection
 EndGlobal
```
Comment (RinoReyns) [inline StabilityClient.Net/StabilityClient.Net.sln:22]: Out of curiosity: in c# you need to add *.sln files to repo? In c++ the good practice is to use cmake that can generate solution/code for any platform if needed.

File: StabilityClient.Net/StabilityClient.Net/IStabilityClient.cs [hunk 5]
```diff
@@ -0,0 +1,10 @@
+using Gooseai;
+
+namespace StabilityClient.Net; 
+
+public interface IStabilityClient {
+    public DashboardService.DashboardServiceClient Dashboard { get; }
+    public EnginesService.EnginesServiceClient Engines { get; }
+    public GenerationService.GenerationServiceClient Generation { get; }
+    public ProjectService.ProjectServiceClient Project { get; }
+}
```
Comment (RinoReyns) [inline StabilityClient.Net/StabilityClient.Net/IStabilityClient.cs]: endline missing

File: StabilityClient.Net/StabilityClient.Net/StabilityClient.cs [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+        return apiKey;
+    }
+    
+    private readonly GrpcChannel _channel;
+
+    private DashboardService.DashboardServiceClient? _dashboard;
+    private EnginesService.EnginesServiceClient? _engines;
+    private GenerationService.GenerationServiceClient? _generation;
+    private ProjectService.ProjectServiceClient? _project;
+}
```
Comment (RinoReyns) [inline StabilityClient.Net/StabilityClient.Net/StabilityClient.cs]: endline is missing

File: StabilityClient.Net/StabilityClient.Net/StabilityClient.cs [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+
+        return apiKey;
+    }
+    
+    private readonly GrpcChannel _channel;
+
+    private DashboardService.DashboardServiceClient? _dashboard;
+    private EnginesService.EnginesServiceClient? _engines;
+    private GenerationService.GenerationServiceClient? _generation;
+    private ProjectService.ProjectServiceClient? _project;
```
Comment (RinoReyns) [inline StabilityClient.Net/StabilityClient.Net/StabilityClient.cs:50]: Again I don't know that is why I'm asking: in C# you keep everything in one file? "Everything" I mean definition of class params and functions?

File: StabilityClient.Net/StabilityClient.Net/StabilityClient.cs [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+                    return Task.CompletedTask;
+                })
+            )
+        };
+
+        _channel = GrpcChannel.ForAddress(host, grpcChannelOptions);
+    }
+
+    public StabilityClient() : this(GetApiKeyFromEnv()) {
+    }
```
Comment (RinoReyns) [inline StabilityClient.Net/StabilityClient.Net/StabilityClient.cs]: This is a constructor of the class?

Comment (Katarzyna-Kadziolka) [reply StabilityClient.Net/StabilityClient.Net/StabilityClient.cs]: Yes, one of two :)

File: StabilityClient.Net/StabilityClient.Net/StabilityClient.cs (hunk 7 above)
Comment (Katarzyna-Kadziolka) [reply StabilityClient.Net/StabilityClient.Net/StabilityClient.cs:50]: Yes, in C# everythings we write in classes

File: StabilityClient.Net/StabilityClient.Net.sln (hunk 4 above)
Comment (Katarzyna-Kadziolka) [reply StabilityClient.Net/StabilityClient.Net.sln:22]: Without a solution, you can't comfortably browse through the projects and nothing external can generate this file

File: .gitignore (hunk 1 above)
Comment (Katarzyna-Kadziolka) [reply .gitignore]: Ok, I added all missed lines :)

**Pull request merged**